### DISCUSSION TITLE
C++ overloaded functions don't seem to complete properly

### DISCUSSION THREAD

**Discussion by colliderman** [hrsh7th/nvim-cmp#1359] [Q&A]
Hi all,

I've only recently started using nvim-cmp and have got it working mostly well with C++, Clangd, and the built in neovim lsp, however I'm having the following issue:

When a function has multiple argument types, the completion window shows something like "2 overloads" for the description. 
![Screen Shot 2022-12-19 at 3 34 51 PM](https://user-images.githubusercontent.com/42239540/208519578-623708ce-f5f8-40fb-865d-2da87ae62ac0.png)

Then when I select the function, the functions are displayed
![Screen Shot 2022-12-19 at 3 35 28 PM](https://user-images.githubusercontent.com/42239540/208519575-e5f3d367-30c9-416b-8d33-f0d65f25e2b7.png)

 but when I select one it doesn't autofill the arguments for me, to allow for easy tabbing through them. 
![Screen Shot 2022-12-19 at 3 35 57 PM](https://user-images.githubusercontent.com/42239540/208519573-87e9e16c-2bd7-43d5-8c2b-73e483713b2e.png)

If a function doesn't have an overload, then when I select it, it auto fills the arguments and I can start typing to fill in the first argument, hit tab, type out the second argument etc. 
![Screen Shot 2022-12-19 at 3 37 10 PM](https://user-images.githubusercontent.com/42239540/208519570-964a59a1-1042-4609-bea8-5205cd2f684d.png)
![Screen Shot 2022-12-19 at 3 37 33 PM](https://user-images.githubusercontent.com/42239540/208519564-d7e053d6-0cb7-4c03-be15-a2fa6c9b3a1a.png)

Is this is expected behaviour or if not, maybe a setting that I haven't set properly?

**Comment by chrisWop** [comment]
I have the same issue.  Has there ever been a solution?

**Comment by alepapadop** [comment]
I confirm that there is an "issue" with the overloads.Propably some manual handling is neede in order to autocomplete the correct part of the helping window. 

A workaround it to pass ""--completion-style=detailed" in the lsp configuration.

<img width="554" height="495" alt="image" src="https://github.com/user-attachments/assets/57a8fb42-24e1-44de-940d-a844e7c94afd" />

**Comment by alepapadop** [comment]
Well I started debuging this issue and it seems it is a clangd issue,  the insertText returned by clangd is not what we expect

In the [<CR>] keymap I inspected the selected item, the code bellow.

<img width="396" height="236" alt="image" src="https://github.com/user-attachments/assets/2fe8fa42-7463-4b66-bac7-ad445fcc6f00" />


The output of this code in the :messages is the following

>    49   detail = "[5 overloads]",
>    50   filterText = "find_first_not_of",
>    51   insertText = "find_first_not_of($0)",
>    52   insertTextFormat = 2,
>    53   kind = 2,
>    54   label = " find_first_not_of",
>    55   labelDetails = {
>    56     detail = "(…)"
>    57   },
>    58   score = 0.93680763244629,
>    59   sortText = "40902d60find_first_not_of",
>    60   textEdit = {
>    61     newText = "find_first_not_of($0)",
>    62     range = {
>    63       ["end"] = {
>    64         character = 6,
>    65         line = 42
>    66       },
>    67       start = {
>    68         character = 6,
>    69         line = 42
>    70       }
>    71     }
>    72   }
>    73 }
>    74 {
>    75   documentation = {
>    76     kind = "markdown",
>    77     value = "find_first_not_of(***const Tp &svt***, size_type pos = 0) -> _If_sv<Tp, size_type>"
>    78   },
>    79   filterText = " ",
>    80   insertText = "const",
>    81   label = "const Tp &svt",
>    82   preselect = true,
>    83   word = ""
>    84 }
>
